src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-page.component.ts

```diff
@@ -283,24 +284,62 @@ export class InstructorSessionResultPageComponent extends InstructorCommentsComp
       // Do not re-fetch data
       return;
     }
-    this.feedbackSessionsService.getFeedbackSessionResults({
-      questionId,
-      courseId: this.session.courseId,
-      feedbackSessionName: this.session.feedbackSessionName,
-      intent: Intent.INSTRUCTOR_RESULT,
-    })
-    .subscribe((resp: SessionResults) => {
-      if (resp.questions.length) {
-        const responses: QuestionOutput = resp.questions[0];
-        this.questionsModel[questionId].responses = responses.allResponses;
-        this.questionsModel[questionId].statistics = responses.questionStatistics;
-        this.questionsModel[questionId].hasPopulated = true;
 
-        this.preprocessComments(responses.allResponses);
-      }
-    }, (resp: ErrorMessageOutput) => {
-      this.statusMessageService.showErrorToast(resp.error.message);
-    });
+    const missingRespMap: Map<string, ResponseOutput> = new Map();
+    const tempMap: Map<string, ResponseOutput> = new Map();
+
+    this.courseService.getCourseSectionNames(this.session.courseId)
```

Comment:
We should not be fetching section names every-time the question tab is pressed. This should be a one-time operation.

Reply:
So we should instead cache the section names and use it for the response fetching instead?

Reply:
Yeah you have a point.. although it feels very unlikely that the section will change while this page is still open. 

It's the same logic as what if someone adds a new question while this page is open - the question will only be shown upon refresh. Plus remember, we only load the results once - you press the tab again and you will get the earlier results :)

The fix above works well for large courses but the tradeoff is the increased number of API calls. For small courses we are making an unusually high amount of API calls every time the button is pressed. I feel section names should be cached as it is a reasonable tradeoff.

Reply:
I see. So it shall be! I think the assumption that sections don't usually change is reasonably, given that it has no real life meaning to the instructor.